### PR TITLE
Allow username change at first login (KEYCLOAK-1857)

### PR DESCRIPTION
**Pull request by dario-campagna**

### CODE REVIEW

File: testsuite/integration/src/test/java/org/keycloak/testsuite/AssertEvents.java [hunk 1]
```diff
@@ -122,6 +122,10 @@ public ExpectedEvent expectRequiredAction(EventType event) {
         return expectLogin().event(event).removeDetail(Details.CONSENT).session(isUUID());
     }
 
+    public ExpectedEvent expectRequiredActionEnabledUsername(EventType event, String newUsername) {
```
Comment (stianst) [inline testsuite/integration/src/test/java/org/keycloak/testsuite/AssertEvents.java]: This is only used once so no need to add it here

File: testsuite/integration/src/test/java/org/keycloak/testsuite/AssertEvents.java [hunk 2]
```diff
@@ -133,6 +137,17 @@ public ExpectedEvent expectLogin() {
                 .session(isUUID());
     }
 
+    public ExpectedEvent expectLogin(String username) {
```
Comment (stianst) [inline testsuite/integration/src/test/java/org/keycloak/testsuite/AssertEvents.java]: Just call "expectLogin().detail(Details.USERNAME, usrname)" to override the expected username, no need to add an additional method

File: testsuite/integration/src/test/java/org/keycloak/testsuite/AssertEvents.java [hunk 3]
```diff
@@ -133,6 +137,17 @@ public ExpectedEvent expectLogin() {
...
+        return expect(EventType.LOGIN, username)
```
Comment (stianst) [inline testsuite/integration/src/test/java/org/keycloak/testsuite/AssertEvents.java]: Again, just use expectLogin(), adding this is not necessary

File: forms/common-themes/src/main/resources/theme/base/login/login-update-profile.ftl [hunk 4]
```diff
@@ -6,6 +6,15 @@
         ${msg("loginProfileTitle")}
     <#elseif section = "form">
         <form id="kc-update-profile-form" class="${properties.kcFormClass!}" action="${url.loginAction}" method="post">
+            <div class="${properties.kcFormGroupClass!} ${messagesPerField.printIfExists('username',properties.kcFormGroupErrorClass!)}" <#if !realm.editUsernameAllowed>style="display:none"</#if>>
```
Comment (stianst) [inline forms/common-themes/src/main/resources/theme/base/login/login-update-profile.ftl]: This should not hide the fields with css, instead the whole section should be omitted from the output when username can't be changed

**Comment by stianst** [comment]
A few comments on the code, also you've referenced the wrong JIRA issue. KEYCLOAK-1849 is the correct one.

**Comment by stianst** [comment]
Closing as I've opened another PR (https://github.com/keycloak/keycloak/pull/1633) with a minor fix (https://github.com/keycloak/keycloak/commit/be102046e6169e0985c93d962b8796a621fd5fd7). The username field wasn't pre-filled with the existing username, or with the update username in the case of an error in the form.

**Comment by dario-campagna** [comment]
Ok, thank you!

**Pull request closed without merging**